src/remoted/manager.c

```diff
@@ -221,12 +222,18 @@ void save_controlmsg(const keyentry * key, char *r_msg, size_t msg_length, int *
                     }
                 }
             }
-         } else {
+        } else {
             /* Update message */
             mdebug2("save_controlmsg(): inserting '%s'", msg);
             os_free(data->message);
             os_strdup(msg, data->message);
 
+            os_free(data->group);
+            if (OS_SUCCESS != lookfor_agent_group(key->id, data->message, &data->group)) {
+                merror("Error getting agent group for agent: '%s'", key->id);
```

Comment:
Don't return at this point. We also need to keep processing the message to save the connection status of the agent in the DB. Change this log error to debug1 and remove the return from the next line.

Reply:
Done.